core/trino-main/src/main/java/io/trino/server/ServerPluginsProvider.java

```diff
@@ -70,10 +71,12 @@ public void loadPlugins(Loader loader, ClassLoaderFactory createClassLoader)
     private static List<File> listFiles(File path)
     {
         try {
-            return stream(newDirectoryStream(path.toPath()))
-                    .map(Path::toFile)
-                    .sorted()
-                    .collect(toImmutableList());
+            try (DirectoryStream<Path> directoryStream = newDirectoryStream(path.toPath())) {
+                return stream(directoryStream)
```

Comment:
You could go one step further and use `Files::list`, which returns a `Stream<Path>`, so that calling `Streams::stream` is not necessary

Reply:
out of scope.